src/test/java/guitests/PersonEditGuiTest.java

```diff
@@ -42,35 +45,116 @@ public void editPerson() {
+        assertTrue(editPersonDialog.isValidEditDialog());
+    }
+
+    @Test
+    public void editPerson_dataValidation() {
+
+        personListPanel.clickOnPerson(td.alice);
+        EditPersonDialogHandle editPersonDialog =  personListPanel.clickEdit();
+        editPersonDialog.enterFirstName("Peter");
+        editPersonDialog.enterLastName("");
+        editPersonDialog.clickOk();
+
+        assertTrue(editPersonDialog.isInputValidationErrorDialogShown());
+        editPersonDialog.dissmissErrorMessage("Invalid Fields");
+        assertFalse(editPersonDialog.isInputValidationErrorDialogShown());
     }
 
-    /* TODO:
-     * Test using 'Edit' context menu
-     * Test using 'Edit' button
-     * Test 'Cancel'
-     * Test 'OK'
-     * Test data validation (just one case)
-     */
+    @Test
+    public void cancelOperation_usingAccelerator() {
+
+        //Delete
+        PersonCardHandle aliceCard = personListPanel.selectCard(td.alice);
+        personListPanel.use_PERSON_DELETE_ACCELERATOR();
+        assertTrue(aliceCard.isShowingGracePeriod("Deleting"));
+
+        personListPanel.use_PERSON_CHANGE_CANCEL_ACCELERATOR();
+        assertFalse(aliceCard.isShowingGracePeriod("Deleting"));
+        assertEquals(statusBar.getText(), "Delete Person [ " + aliceCard.getFirstName() + " "
+                     + aliceCard.getLastName() + " ] was cancelled.");
+
+        //Edit
+        Person newAlice = new PersonBuilder(td.alice.copy()).withFirstName("Alicia").withLastName("Brownstone")
+                .withStreet("Updated street").withCity("Singapore").withPostalCode("123123")
+                .withBirthday("01.01.1979").withGithubUsername("alicebrown123").withTags(td.colleagues, td.friends).build();
+
+        //Get a reference to the card displaying Alice's details
+        PersonCardHandle alicePersonCard = personListPanel.selectCard(td.alice);
+
+        //Edit Alice to change to new values
+        EditPersonDialogHandle editPersonDialog = personListPanel.use_PERSON_EDIT_ACCELERATOR();
+        editPersonDialog.enterNewValues(newAlice);
+        editPersonDialog.pressEnter();
+
+        personListPanel.clickOnPerson(newAlice);
+        assertEquals(alicePersonCard, newAlice);
+        personListPanel.use_PERSON_CHANGE_CANCEL_ACCELERATOR();
+        assertEquals(alicePersonCard, td.alice);
+        assertEquals(statusBar.getText(), "Edit Person [ Alice Brown -> Alicia Brownstone ] was cancelled.");
+
+        //New
+        EditPersonDialogHandle addPersonDialog = personListPanel.clickNew();
+        Person pandaWong = new PersonBuilder("Panda", "Wong")
+                .withStreet("Chengdu Panda Street").withCity("Chengdu").withPostalCode("PANDA")
+                .withBirthday("01.01.1979").withGithubUsername("panda").withTags(td.colleagues, td.friends).build();
+        addPersonDialog.enterNewValues(pandaWong);
+        addPersonDialog.clickOk();
+
+        personListPanel.clickOnListView(); // To ensure shortcut keys work properly.
```

Comment:
From what I remember, this is needed for travis headful mode only. It is a workaround and hence should be hidden (inside GuiRobot, GuiHandle, or GuiTestBase) and automatically executed for shortcuts. We can't expect anybody writing GuiTests to be able to remember when and where to use it.